### PR TITLE
Add setting to remap M1/M2 buttons & other fixes

### LINKED ISSUE #7: M1/M2 remap options
Allow remapping M1/M2 buttons to other keycodes.
Keys: `remap_custom_to_m1_value` & `remap_custom_to_m2_value`

### PR DESCRIPTION
**Pull request by miguelguzmanr**
Closes #7

### CODE REVIEW

File: app/src/main/java/de/langerhans/odintools/ui/composables/Preferences.kt [hunk 1]
```diff
@@ -272,9 +280,83 @@ fun VibrationPreferenceDialog(
...
+                        text = stringResource(id = R.string.remapButton),
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(bottom = 16.dp),
+                        style = MaterialTheme.typography.titleLarge
+                    )
+                    Text(
+                        text = stringResource(id = R.string.pressAnyButton),
+                        style = MaterialTheme.typography.labelLarge,
+                        modifier = Modifier
```
Comment (langerhans) [inline app/src/main/java/de/langerhans/odintools/ui/composables/Preferences.kt:331]: Parameter not needed here

File: app/src/main/java/de/langerhans/odintools/main/MainViewModel.kt [hunk 2]
```diff
@@ -170,6 +170,47 @@ class MainViewModel @Inject constructor(
...
+    }
+
+    fun remapButtonDialogDismissed() {
+        _uiState.update {
+            it.copy(showRemapButtonDialog = false)
+        }
+    }
+
+    private fun getDefaultKeyCode(setting: String): Int {
+        if (setting == "remap_custom_to_m1_value")
```
Comment (langerhans) [inline app/src/main/java/de/langerhans/odintools/main/MainViewModel.kt:190]: Please add braces even to single if statements, I just think it looks cleaner :D

File: app/src/main/java/de/langerhans/odintools/main/MainViewModel.kt [hunk 3]
```diff
@@ -170,6 +170,47 @@ class MainViewModel @Inject constructor(
...
+            return 98
```
Comment (langerhans) [inline app/src/main/java/de/langerhans/odintools/main/MainViewModel.kt:191]: Can we use a KEYCODE_ constants here to make it more obvious what these numbers are?

**Review by langerhans — CHANGES_REQUESTED** [review]
Looking good, just some nitpicks :)

File: app/src/main/java/de/langerhans/odintools/ui/composables/Preferences.kt (hunk 1 above)
Comment (miguelguzmanr) [reply app/src/main/java/de/langerhans/odintools/ui/composables/Preferences.kt]: Yup, thanks!

File: app/src/main/java/de/langerhans/odintools/main/MainViewModel.kt (hunk 3 above)
Comment (miguelguzmanr) [reply app/src/main/java/de/langerhans/odintools/main/MainViewModel.kt]: Great idea. Thank you!

**Comment by miguelguzmanr** [comment]
Applied suggestions from code review.
Thank you!

**Review by langerhans — APPROVED** [review]

**Pull request merged**